stores/metadata.go

```diff
@@ -2176,54 +2175,43 @@ func deleteObjects(tx *gorm.DB, bucket string, path string) (numDeleted int64, _
 	return numDeleted, nil
 }
 
-func invalidateSlabHealthByFCID(tx *gorm.DB, fcids []fileContractID) error {
-	return tx.Exec(`
-	UPDATE slabs SET health_valid = 0 WHERE id in (
-	       SELECT *
-	       FROM (
-	               SELECT slabs.id
-	               FROM slabs
-	               LEFT JOIN sectors se ON se.db_slab_id = slabs.id
-	               LEFT JOIN contract_sectors cs ON cs.db_sector_id = se.id
-	               LEFT JOIN contracts c ON c.id = cs.db_contract_id
-	               WHERE health_valid = 1 AND c.fcid IN (?)
-	       ) slab_ids
-	)
-	               `, fcids).Error
-}
+func invalidateSlabHealthByFCID(ctx context.Context, tx *gorm.DB, fcids []fileContractID) error {
+	if len(fcids) == 0 {
+		return nil
+	}
 
-func (s *SQLStore) invalidateSlabHealthByFCID(ctx context.Context, fcids []fileContractID) error {
 	for {
-		var rowsAffected int64
-		err := s.retryTransaction(func(tx *gorm.DB) error {
-			resp := tx.Exec(`
-UPDATE slabs SET health_valid = 0 WHERE id in (
-	SELECT *
-	FROM (
-		SELECT slabs.id
-		FROM slabs
-		LEFT JOIN sectors se ON se.db_slab_id = slabs.id
-		LEFT JOIN contract_sectors cs ON cs.db_sector_id = se.id
-		LEFT JOIN contracts c ON c.id = cs.db_contract_id
-		WHERE health_valid = 1 AND c.fcid IN (?)
-		LIMIT ?
-	) slab_ids
-)
-		`, fcids, refreshHealthBatchSize)
-			rowsAffected = resp.RowsAffected
-			return resp.Error
-		})
-		if err != nil {
-			return fmt.Errorf("failed to invalidate slab health: %w", err)
-		} else if rowsAffected < refreshHealthBatchSize {
-			return nil // done
+		if resp := tx.Debug().Exec(`
+		UPDATE slabs SET health_valid_until = ? WHERE id in (
+			   SELECT *
+			   FROM (
+					   SELECT DISTINCT slabs.id
+					   FROM slabs
+					   LEFT JOIN sectors se ON se.db_slab_id = slabs.id
+					   LEFT JOIN contract_sectors cs ON cs.db_sector_id = se.id
+					   LEFT JOIN contracts c ON c.id = cs.db_contract_id
+					   WHERE c.fcid IN (?)
+					   LIMIT ?
+			   ) slab_ids
+		)`, time.Now().Add(-time.Second).UnixNano(), fcids, refreshHealthBatchSize); resp.Error != nil {
```

Comment:
`time.Now()` is probably fine? Without subtracting a second? Just to make it a bit easier on the eyes.